aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1486,6 +1487,26 @@ impl AptosVM {
         Ok(())
     }
 
+    /// Check whether the bytecode can be published to mainnet based on the unstable tag in the metadata
+    fn check_unstable_bytecode(&self, modules: &[CompiledModule]) -> VMResult<()> {
+        if self.move_vm.chain_id().is_mainnet() {
+            for module in modules {
+                if let Some(metadata) =
+                    aptos_framework::get_compilation_metadata_from_compiled_module(module)
+                {
+                    if metadata.unstable {
+                        return Err(PartialVMError::new(StatusCode::UNSTABLE_BYTECODE)
+                            .with_message(
+                                "unstable code cannot be published to mainnet".to_string(),
```

Comment:
I'd go with a bit different messages: "code marked unstable is not published on mainnet".

Reply:
done